tools/ptrace.c

```diff
@@ -113,10 +124,27 @@ read_long(long *word, int pid, Elf64_Addr addr)
 int
 read_byte(char *byte, int pid, Elf64_Addr addr)
 {
+  /* Hold last ptrace_peekdata.  */
+  static char tlb_value[sizeof(tlb)];
+
   long value;
-  int ret = ptrace_peekdata(&value, pid, addr);
-  if (!ret)
-    *byte = value & 0xff;
+  int ret;
+
+  /* In case the requested data is cached, access the cache
+   * and return the value there.
+   */
+  if (tlb <= addr && addr < tlb + sizeof(tlb)) {
```

Comment:
And here it should be sizeof(tlb_value), not sizeof(tlb).  The latter is an address-sized entity, whereas we need an value- (i.e. long-) sized entity.